core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -491,20 +495,29 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
      * a database operation to obtain the canonical answer.
      */
     private def pollLocalForResult(
-        docid: DocInfo,
+        namespace: Namespace,
         activationId: ActivationId,
-        promise: Promise[Option[WhiskActivation]])(
-            implicit transid: TransactionId): Unit = {
-        queryActivationResponse(activationId, transid) map {
-            activation => promise.trySuccess { Some(activation) }
-        } onFailure {
+        promise: Promise[WhiskActivation])(
```

Comment:
As discussed in person: Can we try to get rid of these `Promise` passing `Unit` returning methods in favor of `Future` composition?